pkg/asset/installconfig/aws/metadata.go

```diff
@@ -74,21 +76,103 @@ func (m *Metadata) AvailabilityZones(ctx context.Context) ([]string, error) {
 	return m.availabilityZones, nil
 }
 
+// EdgeZones retrieves a list of Local zones for the configured region.
+func (m *Metadata) EdgeZones(ctx context.Context) ([]string, error) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if len(m.edgeZones) == 0 {
+		session, err := m.unlockedSession(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		m.edgeZones, err = localZones(ctx, session, m.Region)
+		if err != nil {
+			return nil, errors.Wrap(err, "getting Local Zones")
+		}
+	}
+
+	return m.edgeZones, nil
+}
+
 // EdgeSubnets retrieves subnet metadata indexed by subnet ID, for
 // subnets that the cloud-provider logic considers to be edge
 // (i.e. Local Zone).
-func (m *Metadata) EdgeSubnets(ctx context.Context) (map[string]Subnet, error) {
+func (m *Metadata) EdgeSubnets(ctx context.Context) (Subnets, error) {
 	err := m.populateSubnets(ctx)
 	if err != nil {
 		return nil, errors.Wrap(err, "error retrieving Edge Subnets")
 	}
 	return m.edgeSubnets, nil
 }
 
+// PopulateEdgeSubnets gather zones and it's attributes, setting the subnet structure
+// with zone fields.
+// TODO: subnet is the structure supported to create the MachineSet by asset generator,
+// review the fields without impacting the API could be one alternative of setting
+// empty subnets.
```

Comment:
I don't follow this comment. Can you provide some more context for the problem?

Reply:
thanks for pointing this out, this TODO mark was intentional to discuss in this review.

I added the `PopulateEdgeSubnets()` to supply complex attributes required by edge compute pool for the zone, like ZoneType and preferred instance type (edge uses it by zone group). Currently, that complex type is available under [`Subnets` structure]()  passed-through argument when creating [`MachineSet()`](https://github.com/openshift/installer/blob/05f1133dd9f9528649a24bcc3470092f69f97f86/pkg/asset/machines/aws/machinesets.go#L19-L20). It was added because Machine pools are created by interacting over the [`zones`(slice of strings)](https://github.com/openshift/installer/blob/05f1133dd9f9528649a24bcc3470092f69f97f86/pkg/asset/machines/aws/machinesets.go#L32), part of the install-config API (https://github.com/openshift/installer/blob/05f1133dd9f9528649a24bcc3470092f69f97f86/pkg/types/aws/machinepool.go#L5-L10) (<- I was preventing to modify this)

I think one approach, would be refact the initial part of [`aws.Machineset()`](https://github.com/openshift/installer/blob/05f1133dd9f9528649a24bcc3470092f69f97f86/pkg/asset/machines/aws/machinesets.go#L19-L20) to simplify that argument handling, avoid duplication and allowing more flexibility to share the metadata populated in the earlier stages. I didn't submit it as it will create more overhead/changes in this PR, I can do it in another PR if you think is useful.

Let me know your thoughts.

Reply:
Patrick, as commented on Slack, I opened the following PR to address that improvement isolated; https://github.com/openshift/installer/pull/7339

But I just realized that I can't use the current version of changing the MachinePool attributes as it will generate changes in Install-Config, not proposed initially.

My plan is to revert things, creating a more limited version.